### ISSUE TITLE
Define a mapping to a Selectors Element Tree

### ISSUE THREAD

**Issue by gsnedders** [whatwg/dom#542]
The Selectors spec has a [Guidance on Mapping Source Documents & Data to an Element Tree](https://drafts.csswg.org/selectors-4/#dom-mapping) appendix, and DOM should define such a mapping according to @tabatkins.

**Comment by annevk** [comment]
That data model does not work though, see, e.g., `:empty`.

**Comment by annevk** [comment]
Looking at that section today I think this is no longer needed @whatwg/css? It seems that Selectors now assumes the DOM and if you have a different host language you need to map it to the DOM. Is that correct?

**Comment by tabatkins** [comment]
Correct. I rebased the Selector model *onto* the DOM, and any other host languages can specify their mapping to the DOM concepts.
